src/gui/speedlimitdialog.h

```diff
@@ -43,17 +47,29 @@ class SpeedLimitDialog : public QDialog
 public:
     explicit SpeedLimitDialog(QWidget *parent);
     ~SpeedLimitDialog();
-    static long askSpeedLimit(QWidget *parent, bool *ok, const QString &title, long defaultVal, long maxVal = 10240000);
+
+public slots:
+    void accept() override;
 
 private slots:
-    void updateSpinValue(int val);
-    void updateSliderValue(int val);
+    void updateUploadSpinValue(int val);
+    void updateDownloadSpinValue(int val);
+    void updateUploadSliderValue(int val);
+    void updateDownloadSliderValue(int val);
+
+    void updateAltUploadSpinValue(int val);
+    void updateAltDownloadSpinValue(int val);
+    void updateAltUploadSliderValue(int val);
+    void updateAltDownloadSliderValue(int val);
 
 private:
-    void setupDialog(long maxSlider, long val);
-    int getSpeedLimit() const;
+    void initDialog();
+    int getUploadSpeedLimit() const;
+    int getDownloadSpeedLimit() const;
+    int getAltUploadSpeedLimit() const;
+    int getAltDownloadSpeedLimit() const;
+
+    BitTorrent::Session *m_session;
```

Comment:
Is this member variable really needed?

Reply:
I thought so but, from this comment, i guess there is no problem using session::instance() where needed.

Reply:
You can declare local variable if you need to access it several times per function.